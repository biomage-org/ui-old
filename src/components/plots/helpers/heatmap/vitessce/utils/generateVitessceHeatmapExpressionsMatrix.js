import _ from 'lodash';
import { convertRange } from 'utils/plotUtils';

const scaledTo255 = (rowOfExpressions, min, max) => (
  rowOfExpressions.map((value) => convertRange(value, [min, max], [0, 255]))
);

const generateVitessceHeatmapExpressionsMatrix = (cellOrder, geneOrder, expressionMatrix) => {
  const geneExpressionsDataMatrix = [];

  geneOrder.forEach((gene) => {
    if (!expressionMatrix.geneIsLoaded(gene)) return;
    const truncatedExpression = expressionMatrix.getTruncatedExpression(gene, cellOrder);

    const { truncatedMin, truncatedMax } = expressionMatrix.getStats(gene);

    const scaledGeneExpressions = scaledTo255(truncatedExpression, truncatedMin, truncatedMax);

    geneExpressionsDataMatrix.push(scaledGeneExpressions);
  });

  return _.flatten(_.unzip(geneExpressionsDataMatrix));
};

export default generateVitessceHeatmapExpressionsMatrix;
