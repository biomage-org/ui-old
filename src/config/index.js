import defaultConfig from 'config/defaultConfig';
import testConfig from 'config/testConfig';

// eslint-disable-next-line import/no-mutable-exports
let config;
if (process.env.NODE_ENV === 'test' || process.env.JEST_WORKER_ID) {
  config = testConfig;
} else {
  config = defaultConfig;
}

export default config;
