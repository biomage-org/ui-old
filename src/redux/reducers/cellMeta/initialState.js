const mitochondrialContent = {
  loading: true,
  error: false,
  data: [],
};

const doubletScores = {
  loading: true,
  error: false,
  data: [],
};

const numOfGenes = {
  loading: true,
  error: false,
  data: [],
};

const numOfUmis = {
  loading: true,
  error: false,
  data: [],
};

const initialState = {
  mitochondrialContent,
  doubletScores,
  numOfGenes,
  numOfUmis,
};

export default initialState;
