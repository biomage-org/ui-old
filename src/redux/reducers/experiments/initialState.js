const experimentTemplate = {
  name: null,
  description: 'Analysis description',
  id: null,
  createdAt: null,
  updatedAt: null,
  notifyByEmail: true,
  meta: { organism: null, type: '10x' },
  sampleIds: [],
  metadataKeys: [],
};

const initialState = {
  meta: {
    loading: false,
    error: false,
    saving: false,
    activeExperimentId: null,
  },
  ids: [],
};

const METADATA_DEFAULT_VALUE = 'N.A.';

export default initialState;
export { experimentTemplate, METADATA_DEFAULT_VALUE };
