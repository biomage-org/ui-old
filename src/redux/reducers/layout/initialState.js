const initialState = {
  windows: {
    direction: 'row',
    first: {
      first: {
        first: 'UMAP', second: 'Cell sets and Metadata', direction: 'row', splitPercentage: 60,
      },
      second: 'Heatmap',
      direction: 'column',
      splitPercentage: 50,
    },
    second: 'Genes',
    splitPercentage: 70,
  },
  panel: 'Gene list',
};

export default initialState;
