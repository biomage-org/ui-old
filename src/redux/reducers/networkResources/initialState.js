const initialState = {
  environment: undefined,
  domainName: undefined,
};

export default initialState;
