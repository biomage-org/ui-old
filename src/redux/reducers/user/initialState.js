const initialState = {
  current: null,
};

export default initialState;
