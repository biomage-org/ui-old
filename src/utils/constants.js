const api = {
  V1: 'v1',
  V2: 'v2',
};

const modules = {
  DATA_MANAGEMENT: 'DATA_MANAGEMENT',
  DATA_PROCESSING: 'DATA_PROCESSING',
  DATA_EXPLORATION: 'DATA_EXPLORATION',
  PLOTS_AND_TABLES: 'PLOTS_AND_TABLES',
  SETTINGS: 'SETTINGS',
  DEFAULT: '',
};

const sampleTech = {
  '10X': '10x',
  RHAPSODY: 'rhapsody',
};

const plotTypes = {
  CONTINUOUS_EMBEDDING: 'ContinuousEmbedding',
  CATEGORICAL_EMBEDDING: 'CategoricalEmbedding',
  HEATMAP: 'Heatmap',
  MARKER_HEATMAP: 'MarkerHeatmap',
  VOLCANO_PLOT: 'VolcanoPlot',
  FREQUENCY_PLOT: 'FrequencyPlot',
  VIOLIN_PLOT: 'violin',
  DOT_PLOT: 'DotPlot',
  TRAJECTORY_ANALYSIS: 'TrajectoryAnalysis',
  NORMALIZED_EXPRESSION_MATRIX: 'NormalizedExpressionMatrix',
  IMG_PLOT: 'ImgPlot',
};

const plotNames = {
  CONTINUOUS_EMBEDDING: 'Continuous Embedding',
  CATEGORICAL_EMBEDDING: 'Categorical Embedding',
  HEATMAP: 'Custom Heatmap',
  MARKER_HEATMAP: 'Marker Heatmap',
  VOLCANO_PLOT: 'Volcano Plot',
  FREQUENCY_PLOT: 'Frequency Plot',
  VIOLIN_PLOT: 'Violin Plot',
  DOT_PLOT: 'Dot Plot',
  TRAJECTORY_ANALYSIS: 'Trajectory Analysis',
  NORMALIZED_EXPRESSION_MATRIX: 'Normalized Expression Matrix',
  IMG_PLOT: 'Image plot'
};

const layout = {
  PANEL_HEADING_HEIGHT: 30,
  PANEL_PADDING: 10,
};

export {
  api,
  modules,
  sampleTech,
  plotTypes,
  plotNames,
  layout,
};
