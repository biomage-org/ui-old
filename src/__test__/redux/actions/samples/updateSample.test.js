import _ from 'lodash';
import configureStore from 'redux-mock-store';
import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';
import thunk from 'redux-thunk';

import updateSample from 'redux/actions/samples/updateSample';
import initialState, { sampleTemplate } from 'redux/reducers/samples/initialState';
import { saveSamples } from 'redux/actions/samples';

import {
  SAMPLES_ERROR, SAMPLES_SAVED, SAMPLES_SAVING, SAMPLES_UPDATE,
} from 'redux/actionTypes/samples';

import config from 'config';
import { api } from 'utils/constants';

jest.mock('redux/actions/samples/saveSamples');

const mockStore = configureStore([thunk]);

describe('updateSample action', () => {
  const mockUuid = 'asd123';

  const mockSample = {
    ...sampleTemplate,
    name: 'test sample',
    uuid: mockUuid,
  };

  const mockState = {
    samples: {
      ...initialState,
      [mockSample.uuid]: mockSample,
    },
  };

  beforeEach(() => {
    jest.clearAllMocks();

    saveSamples.mockImplementation(() => async () => { });

    enableFetchMocks();
    fetchMock.resetMocks();
    fetchMock.doMock();
  });

  it('Works correctly', async () => {
    config.currentApiVersion = api.V2;
    fetchMock.mockResponseOnce(() => Promise.resolve(JSON.stringify({})));

    const sampleDiff = {
      name: 'updated name',
    };

    const store = mockStore(mockState);
    await store.dispatch(updateSample(mockUuid, sampleDiff));

    const actions = store.getActions();
    expect(_.map(actions, 'type')).toEqual([SAMPLES_SAVING, SAMPLES_SAVED, SAMPLES_UPDATE]);
    expect(_.map(actions, 'payload')).toMatchSnapshot();

    expect(fetchMock).toHaveBeenCalledWith(
      'http://localhost:3000/v2/experiments/null/samples/asd123',
      {
        body: JSON.stringify(sampleDiff),
        headers: { 'Content-Type': 'application/json' },
        method: 'PATCH',
      },
    );

    expect(saveSamples).not.toHaveBeenCalled();
  });

  it('Error handling works', async () => {
    config.currentApiVersion = api.V2;

    fetchMock.mockRejectOnce(() => Promise.reject(new Error('Api error')));

    const sampleDiff = {
      name: 'updated name',
    };

    const store = mockStore(mockState);
    await store.dispatch(updateSample(mockUuid, sampleDiff));

    const actions = store.getActions();
    expect(_.map(actions, 'type')).toEqual([SAMPLES_SAVING, SAMPLES_ERROR]);
    expect(_.map(actions, 'payload')).toMatchSnapshot();
  });
});
