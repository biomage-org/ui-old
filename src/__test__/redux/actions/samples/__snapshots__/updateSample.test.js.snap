// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateSample action Error handling works 1`] = `
Array [
  Object {
    "message": "Saving sample...",
  },
  Object {
    "error": "We couldn't save your data.",
  },
]
`;

exports[`updateSample action Works correctly 1`] = `
Array [
  Object {
    "message": "Saving sample...",
  },
  undefined,
  Object {
    "sample": Object {
      "name": "updated name",
    },
    "sampleUuid": "asd123",
  },
]
`;
